### PR TITLE
Add warning if parent token's TTL is shorter than its child token

### PR DESCRIPTION
**Pull request by vishalnayak**

### CODE REVIEW

**Comment by jefferai** [comment]
What about the parent's parent? 😭

File: vault/token_store.go [hunk 1]
```diff
@@ -1162,6 +1162,13 @@ func (ts *TokenStore) handleCreateCommon(
 		}
 	}
 
+	// If the token being generated is not an orphan token and if the parent token's
+	// TTL is shorter than its child token, add a warning that the child token will
+	// be revoked if the parent token is revoked.
+	if te.Parent != "" && parent != nil && parent.TTL != 0 && parent.TTL < te.TTL {
+		resp.AddWarning("parent token's TTL is shorter; token will be revoked if the parent token expires due to non-renewal")
+	}
```
Comment (jefferai) [inline vault/token_store.go]: This should start capitalized as it's a warning

**Comment by vishalnayak** [comment]
That would have been displayed when the parent was created :)

**Comment by jefferai** [comment]
And the creator of that token may have been someone else...

File: vault/token_store.go (hunk 1 above)
Comment (vishalnayak) [inline vault/token_store.go]: Done.

**Comment by vishalnayak** [comment]
@jefferai The warning will be added only if the `Parent` is set in the TokenEntry, and only if the parent is non-root with its TTL shorter than the token being generated. I thought it is only when `Parent` is set that revocation of child tokens will happen.

If you feel that this can cause more confusion than help, we can close this.

**Comment by jefferai** [comment]
My issue is really that you may give users a false sense of security if they don't see this warning but further up the chain is a parent that has a shorter TTL. I think if we are going to do this, we need to go all the way up the chain to a token with no parent and check TTL at each point. In many cases this will be a single hop but not always.

**Comment by vishalnayak** [comment]
Oh, I see. The problem I tried to address was to reduce the confusion created for the users of child tokens when parent token skips its renewal. I agree that there is no point in reminding the token generator to renew the parent token in time, the purpose of `token_duration` is just that anyways. It makes sense to not return this warning. We can close this PR.

Alternatively, we could log the revocation of child tokens. Log reflects the revocation of parent token but not the revocation of child tokens. Thoughts?

**Comment by jefferai** [comment]
Don't close it yet, let me think about this some more.

**Comment by vishalnayak** [comment]
This was a wrong idea to start with. Closing it as it is not aiding any use case. Also, it is incorrect to just look at a parent token and set a warning on the child token, since the parent token's TTL can be changed upon renewal, making the warning incorrect/insignificant.

**Pull request closed without merging**